sstables/index_reader.hh

```diff
@@ -711,8 +711,13 @@ private:
             // i is valid until next allocation point
             auto& entries = bound.current_list->_entries;
             if (i == std::end(entries)) {
-                sstlog.trace("index {}: not found", fmt::ptr(this));
-                return advance_to_page(bound, summary_idx + 1);
+                if (_single_page_read) {
+                    sstlog.trace("index {}: not found in index page {}, returning eof because this is a single-partition read", summary_idx, fmt::ptr(this));
+                    return advance_to_end(bound);
```

Comment:
Doesn't it mean that upper bound estimation when it hits the advance_to_next_partition() path will now be worse, because it extends to the end of the data file?

Reply:
"Bound estimation"? We can only enter this code path if we are querying an absent partition key.

Correct me if I'm wrong, but for a single-partition reader, `advance_to` is called only once: for the lower bound, during initialization, from `advance_lower_and_check_if_present`. If we get into this code path, we will set EOF, `advance_lower_and_check_if_present` will notice this, the read will end immediately, and will be reported as a false bloom filter positive. There is nothing to estimate, the Data file isn't touched.

Reply:
`advance_to_next_partition` doesn't call `advance_to` (the function we are modifying), it's a simpler operation.

Reply:
> `advance_to_next_partition` doesn't call `advance_to` (the function we are modifying).

Right, so the comment is moot.